http-server/src/main/java/io/airlift/http/server/RejectForwardedRequestCustomizer.java

```diff
@@ -0,0 +1,28 @@
+package io.airlift.http.server;
+
+import org.eclipse.jetty.http.BadMessageException;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.Request;
+
+public class RejectForwardedRequestCustomizer
+        implements HttpConfiguration.Customizer
+{
+    private static final String X_FORWARDED_PREFIX = "x-forwarded-";
+
+    @Override
+    public Request customize(Request request, HttpFields.Mutable responseHeaders)
+    {
+        if (request.getHeaders().stream().anyMatch(RejectForwardedRequestCustomizer::isForwardingHeader)) {
+            throw new BadMessageException(HttpStatus.NOT_ACCEPTABLE_406, "Forwarded header is forbidden");
+        }
+        return request;
+    }
+
+    private static boolean isForwardingHeader(HttpField httpField)
+    {
+        return httpField.getLowerCaseName().startsWith(X_FORWARDED_PREFIX);
```

Comment:
`httpField.getName().regionMatches(true, 0, X_FORWARDED_PREFIX, 0, X_FORWARDED_PREFIX.length())` might be more efficient, by avoiding lowercasing (`getLowerCaseName` returns precomputed value for some headers, but not all)

Reply:
Nice!